vllm/worker/worker.py

```diff
@@ -181,14 +185,26 @@ def initialize_cache(self, num_gpu_blocks: int,
 
     def _init_cache_engine(self):
         assert self.cache_config.num_gpu_blocks is not None
-        self.cache_engine = CacheEngine(self.cache_config, self.model_config,
-                                        self.parallel_config)
+        with CudaMemoryProfiler() as m:
+            self.cache_engine = CacheEngine(self.cache_config,
+                                            self.model_config,
+                                            self.parallel_config)
+        mem_usage = m.consumed_memory
+        unit, scale = "GB", float(2**30)
+        logger.info("GPU KV cache reserves %.4f %s GPU memory.",
+                    mem_usage / scale, unit)
         self.gpu_cache = self.cache_engine.gpu_cache
         self.model_runner.set_block_size(self.cache_engine.block_size)
 
     def _warm_up_model(self) -> None:
         if not self.model_config.enforce_eager:
-            self.model_runner.capture_model(self.gpu_cache)
+            with CudaMemoryProfiler() as m:
+                self.model_runner.capture_model(self.gpu_cache)
+            mem_usage = m.consumed_memory
+            unit, scale = "GB", float(2**30)
+            logger.info("Capturing cuda graph reserves %.4f %s GPU memory.",
```

Comment:
Should we use `debug` instead?

Reply:
feel like it'd be useful to show exact memory allocation since I've seen some users not understanding this part (like here; https://www.reddit.com/r/LocalLLaMA/comments/1bz3bn1/whats_up_with_vllm/)